minio/credentials/providers.py

```diff
@@ -387,10 +382,10 @@ def fetch(self, url):
 
         res = _urlopen(self._http_client, "GET", url)
         data = json.loads(res.data)
-        if data["Code"] != "Success":
+        if data.get("Code", "Success") != "Success":
```

Comment:
Maybe let's check for   "AccessKeyId", "SecretAccessKey" and "Token" fields, if they are non empty then we can assume this is successful call

Reply:
If those fields don't have proper values, it will fail at line no 393. I think this check is good enough to address other cases. However this fix is only testable in AWS ECS env.